ui/src/BulkUploads.js

```diff
@@ -0,0 +1,322 @@
+    const authorisedActivities = await this.getAuthorisedActivities(); // Only need to do this once; don't refresh it repeatedly as it changes infrequently
+    this.refreshDataFromBackend(authorisedActivities);
+
+    this.interval = setInterval(
+      () => this.refreshDataFromBackend(authorisedActivities),
+      1000
+    );
+  };
+
+  refreshDataFromBackend = async (authorisedActivities) => {
+    if (!authorisedActivities.includes("VIEW_BULK_REFUSAL_PROGRESS")) return;
+
+    const response = await fetch(
+      `/api/job?collectionExercise=${this.props.collectionExerciseId}&jobType=BULK_REFUSAL`
+    );
+    const bulkRefusalJobs = await response.json();
+
+    this.setState({ bulkRefusalJobs: bulkRefusalJobs });
+  };
+
+  getAuthorisedActivities = async () => {
+    const response = await fetch(`/api/auth?surveyId=${this.props.surveyId}`);
+
+    // TODO: We need more elegant error handling throughout the whole application, but this will at least protect temporarily
+    if (!response.ok) {
+      return;
+    }
+
+    const authJson = await response.json();
+
+    this.setState({ authorisedActivities: authJson });
+
+    return authJson;
+  };
+
+  handleBulkRefusalUpload = (e) => {
+    if (e.target.files.length === 0) {
+      return;
+    }
+
+    // Display the progress modal dialog
+    this.setState({
+      uploadInProgress: true,
+    });
+
+    const formData = new FormData();
+    formData.append("file", e.target.files[0]);
+
+    const fileName = e.target.files[0].name;
+    // Reset the file
+    e.target.value = null;
+
+    // Send the file data to the backend
+    axios
+      .request({
+        method: "post",
+        url: "/api/upload",
+        data: formData,
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+        onUploadProgress: (p) => {
+          console.log(p);
+
+          // Update file upload progress
+          this.setState({
+            fileProgress: p.loaded / p.total,
+          });
+        },
+      })
+      .then((data) => {
+        // send the job details
+        const fileId = data.data;
+        const jobData = new FormData();
+        jobData.append("fileId", fileId);
+        jobData.append("fileName", fileName);
+        jobData.append("collectionExerciseId", this.props.collectionExerciseId);
+        jobData.append("jobType", "BULK_REFUSAL");
+
+        const response = fetch(`/api/job`, {
+          method: "POST",
+          body: jobData,
+        });
+
+        if (!response.ok) {
+          // TODO - nice error handling
+          // If we check it, it's is currently buggy and leaves the popup on the screen for unknown reasons - need to raise a defect
+        }
+
+        // Hide the progress dialog and flash the snackbar message
+        this.setState({
+          fileProgress: 1.0,
+          fileUploadSuccess: true,
+          uploadInProgress: false,
+        });
+
+        this.getBulkRefusalJobs();
```

Comment:
I think you should get rid of this and leave it to `refreshDataFromBackend` to update the table